trace-reader/src/loader.rs

```diff
@@ -0,0 +1,351 @@
+use std::{
+    env,
+    fmt::Debug,
+    fs::File,
+    io::{Error, ErrorKind, Read, Seek, SeekFrom},
+    mem::size_of,
+    usize,
+};
+
+use common::Intensity;
+
+#[derive(Default, Debug)]
+pub struct TraceFileHeader {
+    pub prog_version: String,
+    pub run_descript: String,
+    pub resolution: i32,
+    pub number_of_channels: i32,
+    pub channel_enabled: Vec<bool>,
+    pub volts_scale_factor: Vec<f64>,
+    pub channel_offset_volts: Vec<f64>,
+    pub sample_time: f64,
+    pub number_of_samples: i32,
+    pub trigger_enabled: Vec<bool>,
+    pub ex_trigger_enabled: bool,
+    pub trigger_level: Vec<f64>,
+    pub ex_trigger_level: f64,
+    pub trigger_slope: Vec<i32>,
+    pub ex_trigger_slope: i32,
+    total_bytes: usize,
+}
+
+impl TraceFileHeader {
+    pub fn load(file: &mut File) -> Result<Self, Error> {
+        let mut total_bytes = usize::default();
+        let prog_version = load_string(file, &mut total_bytes)?;
+        let run_descript = load_string(file, &mut total_bytes)?;
+        let resolution = load_i32(file, &mut total_bytes)?;
+        let number_of_channels = load_i32(file, &mut total_bytes)?;
+        Ok(TraceFileHeader {
+            prog_version,
+            run_descript,
+            resolution,
+            number_of_channels,
+            channel_enabled: load_bool_vec(file, number_of_channels as usize, &mut total_bytes)?,
+            volts_scale_factor: load_f64_vec(file, number_of_channels as usize, &mut total_bytes)?,
+            channel_offset_volts: load_f64_vec(
+                file,
+                number_of_channels as usize,
+                &mut total_bytes,
+            )?,
+            sample_time: load_f64(file, &mut total_bytes)?,
+            number_of_samples: load_i32(file, &mut total_bytes)?,
+            trigger_enabled: load_bool_vec(file, number_of_channels as usize, &mut total_bytes)?,
+            ex_trigger_enabled: load_bool(file, &mut total_bytes)?,
+            trigger_level: load_f64_vec(file, number_of_channels as usize, &mut total_bytes)?,
+            ex_trigger_level: load_f64(file, &mut total_bytes)?,
+            trigger_slope: load_i32_vec(file, number_of_channels as usize, &mut total_bytes)?,
+            ex_trigger_slope: load_i32(file, &mut total_bytes)?,
+            total_bytes,
+        })
+    }
+}
+
+impl TraceFileHeader {
+    fn get_total_bytes(&self) -> usize {
+        self.total_bytes
+    }
```

Comment:
Please leave a blank line between function blocks, not doing so hurts readability.